src/miner_keys.erl

```diff
@@ -128,27 +198,68 @@ key_config() ->
         KC -> KC
     end.
 
+-spec libp2p_to_gateway_key(libp2p_crypto:key_map()) -> libp2p_crypto:key_map().
+libp2p_to_gateway_key(#{secret := {ecc_compact, PrivKey}}) ->
+    {ok, #{
+        secret => {ecc_compact, PrivKey#'ECPrivateKey'{publicKey = <<>>}},
+        public => {ecc_compact, <<>>}
+    }}.
+
 %% @doc prints the public hotspot and onboadring key in a file:consult
 %% friendly way to stdout. This is used by other services (like
 %% gateway_config) to get read access to the public keys
 print_keys(_) ->
     #{
-       pubkey := PubKey,
-       onboarding_key := OnboardingKey
-     } = keys(),
-    MaybeB58 = fun(undefined) -> undefined;
-                  (Key) -> libp2p_crypto:pubkey_to_b58(Key)
-               end,
-    Props = [{pubkey, MaybeB58(PubKey)},
-             {onboarding_key, OnboardingKey}
-            ] ++ [ {animal_name, element(2, erl_angry_purple_tiger:animal_name(libp2p_crypto:pubkey_to_b58(PubKey)))} || PubKey /= undefined ],
+        pubkey := PubKey,
+        onboarding_key := OnboardingKey
+    } = keys(),
+    MaybeB58 = fun
+        (undefined) -> undefined;
+        (Key) -> libp2p_crypto:pubkey_to_b58(Key)
+    end,
+    Props =
+        [
+            {pubkey, MaybeB58(PubKey)},
+            {onboarding_key, OnboardingKey}
+        ] ++
+            [
+                {animal_name,
+                    element(
+                        2, erl_angry_purple_tiger:animal_name(libp2p_crypto:pubkey_to_b58(PubKey))
+                    )}
+             || PubKey /= undefined
+            ],
     lists:foreach(fun(Term) -> io:format("~tp.~n", [Term]) end, Props),
     rpc_ok.
 
+key_proplist_to_uri(Props) ->
+    Bus = proplists:get_value(bus, Props, "i2c-1"),
+    Address = proplists:get_value(address, Props, 16#60),
+    KeySlot = proplists:get_value(key_slot, Props, 0),
+    Network = application:get_env(blockchain, network, mainnet),
+    PortAndQuery = lists:flatten(io_lib:format(":~p?slot=~p&network=~s", [Address, KeySlot, Network])),
+    "ecc://" ++ Bus ++ PortAndQuery.
```

Comment:
Why not do this in the format string on the previous line?

Reply:
i think i was probably using `~p` there and getting the extra escaped string quotes in the result; forgot to collapse these when i added the network which is also either a string or an atom